controller/contour/deployment.go

```diff
@@ -0,0 +1,345 @@
+			}
+			return nil
+		}
+		return fmt.Errorf("failed to get deployment %s/%s: %w", desired.Namespace, desired.Name, err)
+	}
+
+	if err := r.updateDeploymentIfNeeded(ctx, current, desired); err != nil {
+		return fmt.Errorf("failed to update deployment %s/%s: %w", desired.Namespace, desired.Name, err)
+	}
+
+	return nil
+}
+
+// ensureDeploymentDeleted ensures the deployment for the provided contour
+// is deleted.
+func (r *Reconciler) ensureDeploymentDeleted(ctx context.Context, contour *operatorv1alpha1.Contour) error {
+	deployment := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: contour.Spec.Namespace.Name,
+			Name:      contour.Name,
+		},
+	}
+
+	if err := r.Client.Delete(ctx, deployment); err != nil {
+		if errors.IsNotFound(err) {
+			return nil
+		}
+		return err
+	}
+
+	r.Log.Info("deleted deployment", "namespace", deployment.Namespace, "name", deployment.Name)
+	return nil
+}
+
+// desiredDeployment returns the desired deployment for the provided contour using
+// image as Contour's container image.
+func desiredDeployment(contour *operatorv1alpha1.Contour, image string) (*appsv1.Deployment, error) {
+	parsedImage := strings.Split(image, ":")
+	labels := map[string]string{
+		"app.kubernetes.io/name":     "contour",
+		"app.kubernetes.io/instance": contour.Name,
+		// The image tag is used as the version.
+		"app.kubernetes.io/version":    parsedImage[1],
+		"app.kubernetes.io/component":  "ingress-controller",
+		"app.kubernetes.io/managed-by": "contour-operator",
+		// Associate the deployment with the provided contour.
+		owningContourLabel: contour.Name,
+	}
+
+	container := corev1.Container{
+		Name:            contourContainerName,
+		Image:           image,
+		ImagePullPolicy: corev1.PullIfNotPresent,
+		Command: []string{
+			"contour",
+		},
+		Args: []string{
+			"serve",
+			"--incluster",
+			"--xds-address=0.0.0.0",
+			"--xds-port=8001",
+			"--envoy-service-http-port=80",
+			"--envoy-service-https-port=443",
+			fmt.Sprintf("--contour-cafile=%s", filepath.Join("/", certsVolumeMountDir, "ca.crt")),
+			fmt.Sprintf("--contour-cert-file=%s", filepath.Join("/", certsVolumeMountDir, "tls.crt")),
+			fmt.Sprintf("--contour-key-file=%s", filepath.Join("/", certsVolumeMountDir, "tls.key")),
+			fmt.Sprintf("--config-path=%s", filepath.Join("/", cfgVolumeMountDir, cfgFileName)),
+		},
+		Env: []corev1.EnvVar{
+			{
+				Name: deployNsEnvVar,
+				ValueFrom: &corev1.EnvVarSource{
+					FieldRef: &corev1.ObjectFieldSelector{
+						APIVersion: "v1",
+						FieldPath:  "metadata.namespace",
+					},
+				},
+			},
+			{
+				Name: deployPodEnvVar,
+				ValueFrom: &corev1.EnvVarSource{
+					FieldRef: &corev1.ObjectFieldSelector{
+						APIVersion: "v1",
+						FieldPath:  "metadata.name",
+					},
+				},
+			},
+		},
+		Ports: []corev1.ContainerPort{
+			{
+				Name:          "xds",
+				ContainerPort: 8001,
+				Protocol:      "TCP",
+			},
+			{
+				Name:          "debug",
+				ContainerPort: 8000,
```

Comment:
There really isn't a need to expose this port.

Reply:
@stevesloka the port is being exposed in https://raw.githubusercontent.com/projectcontour/contour/release-1.9/examples/render/contour.yaml, so I'd like to keep it for consistency. I created https://github.com/projectcontour/contour-operator/issues/46 to capture conditionally exposing the debug port.